grammar.js

```diff
@@ -1186,8 +1186,8 @@ function Value(scope, parent) {
 }
 
 Value.prototype.next = function next(type, space, text, scanner) {
-    if (type === 'number') {
-        return this.parent.return(this.scope, ['val', +text], scanner);
+    if (type === 'number' || type === 'string') {
+        return this.parent.return(this.scope, ['val', type === 'number'? +text : text], scanner);
```

Comment:
I think I would prefer to have separate cases for `number` and `string`, even if they are very copy-paste, than have a ternary expression inside this case.